Update jsonUtils.ts

wernerstucky opened the pull request:
Proposed fix for Issue : #160 converting a non-JSON string to Object or Array. This brings back functionality that was present in the classic JSONEditor We catch JSON parse errors and return a reasonable Object / Array.  This is an improvement on the classic editor which used to purge the string value.

josdejong commented:
Thanks @wernerstucky , I like your PR and inserting the original value as array item or object value 👌.

I have a few feedbacks:

1. Naming the new key `"0"` sounds a bit odd to me. How about naming it `"value"` or `"key"` or `"New prop"` or something like that?
2. Can you please write unit tests to validate the new behavior?
3. Does this new behavior also work for numbers, booleans and null? If not, can you make that work too?

wernerstucky commented:
Thanks @josdejong  I'll like those ideas. Let me work on that and we take it from there.

wernerstucky commented:
@josdejong over to you, 

1. value gets added under "value"
2. added a few tests for various values
3. added default return to both array and object cases which will handle boolean, number, etc

josdejong commented:
Thanks for the updates, this looks good 👌 

Can you please run `npm run format` to fix the linting issues? (you can probably configure your IDE to apply formatting when saving a file).

wernerstucky commented:
Looks like those linting issues are fixed.  I've installed the Prettier extension thanks for the tip!

josdejong commented:
💪 thanks!

josdejong commented:
Published now in `v0.7.11`
